odsclient/core.py

```diff
@@ -251,6 +256,32 @@ def get_whole_dataset(self,
 
         return result
 
+    # noinspection PyShadowingBuiltins
+    def push_dataset_realtime(self,
+                              dataset_id,         # type: str
+                              payload,            # type: List[Dict[str,Any]]
+                              push_key,           # type: str
+                              **other_opts
+                              ):
+        """
+        Pushes a Dataset.
+
+        :param dataset_id:
+        :param payload: The dataset to push as a list of dists, where the dict keys are the column names
+        :param push_key: The Push Key provided by the API for pushing this dataset
+        :returns: HTTP Response status
+        """
+
+        # Combine all the options
+        opts = other_opts
+        opts['pushkey'] = push_key
```

Comment:
Rather `opts['pushkey']= self.get_apikey()` to mimic the behaviour in the get api ?

Reply:
See my previous comment. I've started out like this but it seems you need a dedicated push key for each dataset, which is independent of the api_key.

Reply:
ok